File: de.lmu.ifi.dbs.medmon.algorithm.extension/src/de/lmu/ifi/dbs/medmon/algorithm/activator/Activator.java
package de.lmu.ifi.dbs.medmon.algorithm.activator;

import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;
import org.osgi.util.tracker.ServiceTracker;

import de.lmu.ifi.dbs.medmon.algorithm.extension.ISensorDataAlgorithm;

public class Activator implements BundleActivator {

	private static ServiceTracker algorithmTracker; 
	
	@Override
	public void start(BundleContext context) throws Exception {
		algorithmTracker = new ServiceTracker(context, ISensorDataAlgorithm.class.getName(), null);
		algorithmTracker.open();
	}

	@Override
	public void stop(BundleContext context) throws Exception {
		algorithmTracker.close();
	}
	
	public static ISensorDataAlgorithm[] getAlgorithmServices() {
		return (ISensorDataAlgorithm[]) algorithmTracker.getServices();
	}

}

File: de.lmu.ifi.dbs.knowing.core.common/src/de/lmu/ifi/dbs/knowing/core/common/internal/Activator.scala
package de.lmu.ifi.dbs.knowing.core.common.internal

import org.osgi.framework.{BundleActivator,BundleContext}
import de.lmu.ifi.dbs.knowing.core.util.OSGIUtil
import de.lmu.ifi.dbs.knowing.core.common._
import de.lmu.ifi.dbs.knowing.core.common.io._

/**
 * @author Nepomuk Seiler
 * 
 */
class Activator extends BundleActivator {

  private var osgiUtil: OSGIUtil = _
  
  def start(context: BundleContext) = {
     osgiUtil = new OSGIUtil(context)
     osgiUtil.registerProcessor(new AddClassAttributeFactory)
     osgiUtil.registerSaver(new InstancesSaverFactory)
     osgiUtil.registerLoader(new InstancesLoaderFactory)
     osgiUtil.registerProcessor(new UncertainClassDumperFactory)
  }

  def stop(context: BundleContext) = {
    osgiUtil.deregisterAll
    osgiUtil = null
  }

}